File: CS-Examples/26_StructuredDocumentTag/RemoveContentControls.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;
using System.Drawing;

namespace RemoveContentControls
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            
			// Create a new document object
			Document doc = new Document();

			// Load a document file from a specified path
			doc.LoadFromFile(@"...\..\..\..\..\..\Data\RemoveContentControls.docx");

			// Iterate through the sections in the document
			for (int s = 0; s < doc.Sections.Count; s++)
			{
				// Get the current section
				Section section = doc.Sections[s];

			// Iterate through the child objects in the section's body
			for (int i = 0; i < section.Body.ChildObjects.Count; i++)
			{
				// Check if the child object is a paragraph
				if (section.Body.ChildObjects[i] is Paragraph)
				{
					// Get the paragraph object
					Paragraph para = section.Body.ChildObjects[i] as Paragraph;
					
					// Iterate through the child objects in the paragraph
					for (int j = 0; j < para.ChildObjects.Count; j++)
					{
						// Check if the child object is a StructureDocumentTagInline
						if (para.ChildObjects[j] is StructureDocumentTagInline)
						{
							// Get the StructureDocumentTagInline object
							StructureDocumentTagInline sdt = para.ChildObjects[j] as StructureDocumentTagInline;
							
							// Remove the StructureDocumentTagInline from the paragraph
							para.ChildObjects.Remove(sdt);
							
							// Decrement the index to account for the removed object
							j--;
						}
					}
				}
				
				// Check if the child object is a StructureDocumentTag
				if (section.Body.ChildObjects[i] is StructureDocumentTag)
				{
					// Get the StructureDocumentTag object
					StructureDocumentTag sdt = section.Body.ChildObjects[i] as StructureDocumentTag;
					
					// Remove the StructureDocumentTag from the section's body
					section.Body.ChildObjects.Remove(sdt);
					
					// Decrement the index to account for the removed object
					i--;
				}
			}
			}

			// Save the modified document to a new file
			string output = "RemoveContentControls_out.docx";
			doc.SaveToFile(output, FileFormat.Docx2013);

			// Dispose the document object
			doc.Dispose();

            //Launch the file
            WordDocViewer(output);
        }      
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/26_StructuredDocumentTag/RemoveContentControls.vb
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace RemoveContentControls
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new document object
			Dim doc As New Document()

			' Load a document file from a specified path
			doc.LoadFromFile("..\..\..\..\..\..\Data\RemoveContentControls.docx")

			' Iterate through the sections in the document
			For s As Integer = 0 To doc.Sections.Count - 1
				' Get the current section
				Dim section As Section = doc.Sections(s)

				' Iterate through the child objects in the section's body
				Dim i As Integer = 0
				Do While i < section.Body.ChildObjects.Count
					' Check if the child object is a paragraph
					If TypeOf section.Body.ChildObjects(i) Is Paragraph Then
						' Get the paragraph object
						Dim para As Paragraph = TryCast(section.Body.ChildObjects(i), Paragraph)

						' Iterate through the child objects in the paragraph
						Dim j As Integer = 0
						Do While j < para.ChildObjects.Count
							' Check if the child object is a StructureDocumentTagInline
							If TypeOf para.ChildObjects(j) Is StructureDocumentTagInline Then
								' Get the StructureDocumentTagInline object
								Dim sdt As StructureDocumentTagInline = TryCast(para.ChildObjects(j), StructureDocumentTagInline)

								' Remove the StructureDocumentTagInline from the paragraph
								para.ChildObjects.Remove(sdt)

								' Decrement the index to account for the removed object
								j -= 1
							End If
							j += 1
						Loop
					End If

					' Check if the child object is a StructureDocumentTag
					If TypeOf section.Body.ChildObjects(i) Is StructureDocumentTag Then
						' Get the StructureDocumentTag object
						Dim sdt As StructureDocumentTag = TryCast(section.Body.ChildObjects(i), StructureDocumentTag)

						' Remove the StructureDocumentTag from the section's body
						section.Body.ChildObjects.Remove(sdt)

						' Decrement the index to account for the removed object
						i -= 1
					End If
					i += 1
				Loop
			Next s

			' Save the modified document to a new file
			Dim output As String = "RemoveContentControls_out.docx"
			doc.SaveToFile(output, FileFormat.Docx2013)

			' Dispose the document object
			doc.Dispose()

			'Launch the file
			WordDocViewer(output)
		End Sub
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
